src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -23,8 +25,23 @@ public CommandResult execute() throws CommandException {
         if (!undoRedoStack.canUndo()) {
             throw new CommandException(MESSAGE_FAILURE);
         }
-        undoRedoStack.popUndo().undo();
-        return new CommandResult(MESSAGE_SUCCESS);
+
+        UndoableCommand commandToUndo = undoRedoStack.popUndo();
+
+        Person studentModified = commandToUndo.undo();
+
+        if (studentModified != null && !model.getFilteredPersonList().contains(studentModified)) {
```

Comment:
maybe can create another method in model to make `!model.getFilteredPersonList().contains(studentModified)` obey SLAP